packages/core/src/util/getPreloadFlags.js

```diff
@@ -0,0 +1,21 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+exports.getPreloadFlags = function getPreloadFlags() {
+  if (
+    process.env.NODE_OPTIONS?.includes('--require')
+    || process.env.NODE_OPTIONS?.includes('--import')
+    || process.env.NODE_OPTIONS?.includes('--experimental-loader')) {
+      return process.env.NODE_OPTIONS;
```

Comment:
NODE_OPTIONS can contain [multiple information.](https://nodejs.org/api/cli.html#node_optionsoptions).

Possible update:

```js
const flags = ['--require', '--import', '--experimental-loader'];
  
  if (process.env.NODE_OPTIONS) {
    const relevantFlags = process.env.NODE_OPTIONS.split(' ').filter(option =>
      flags.some(flag => option.includes(flag))
    );
    if (relevantFlags.length > 0) {
      return relevantFlags.join(' ');
    }
  }
  
  if (process.execArgv.length > 0) {
    const result = process.execArgv.find(arg =>
      flags.some(flag => arg.includes(flag))
    );
    return result || 'noFlags';
  } else {
    return 'noFlags';
  }
  ````

Reply:
> NODE_OPTIONS can contain [multiple information.](https://nodejs.org/api/cli.html#node_optionsoptions).
>

Thanks for sharing.  I will update the same according to this suggestion 👍

Reply:
since Array.prototype.some() function returns a boolean, we might have to make some changes in the returning string if the whole argument list is required in the debug logs. 
But as you pointed out, the arguments and NODE_OPTIONS can contain more information that might not be relevant to us in the debug log.
Let me think of other options.

Reply:
Any solution works :)

Reply:
Instead of printing out the whole, now logging only the flags (--import, --require, etc)
Also added a check for `@instana` keyword which will be sufficient to verify that one/some of instana packages are passed in the argument list or as NODE_OPTIONS